### PR TITLE
Adding quick-view options to list view

### LINKED ISSUE #449: [FEATURE] show quick-view options for VMs?
Explore ways to show quick-view actions (either all the actions, or  most commonly used action start/stop/...)
Either as a general component; or specifically for VM (and other user facing views like storage, network)

### PR DESCRIPTION
**Pull request by davidjumani**
Fixes #449

### CODE REVIEW

**Comment by shwstppr** [comment]
@blueorangutan package

File: src/components/view/ListView.vue [hunk 1]
```diff
@@ -60,13 +60,22 @@
     </div>
     -->
 
+    <p v-if="actions.length > 0" :slot="actions.length > 0 ? 'expandedRowRender' : ''" slot-scope="record">
+      {{ record.id }}
+      <action-button
+        size="default"
+        :actions="$route.meta.actions"
+        :dataView="true"
+        :resource="record"
```
Comment (rhtyd) [inline src/components/view/ListView.vue:69]: @davidjumani for some reason only the first row of the table is used when records are clicked/passed -> perhaps need refactoring - let's discuss next week

Comment (davidjumani) [reply src/components/view/ListView.vue:69]: That was the case, so I changed the handleSubmit in Autogen.vue to read thre resource from the currentAction if present

Comment (davidjumani) [reply src/components/view/ListView.vue:69]: Changes made. Can test next week!

**Comment by davidjumani** [comment]
@blueorangutan package

**Comment by davidjumani** [comment]
@blueorangutan package

**Comment by davidjumani** [comment]
@rhtyd @shwstppr good to go!

**Comment by rhtyd** [comment]
@davidjumani can you fix the conflict?

**Comment by davidjumani** [comment]
@blueorangutan package

**Comment by davidjumani** [comment]
@blueorangutan package

**Comment by davidjumani** [comment]
@rhtyd Good to go!

**Comment by rhtyd** [comment]
Subjective/TBD - should we show labels for actions like legacy UI; should the quick view be in a pop/hover-over menu component? Or an icon you click (say the 3-dot icon, similar to chrome, android and other UXs) which shows the quick-view

**Comment by rhtyd** [comment]
Please merge latest master and fix any conflicts? Thanks.

**Comment by davidjumani** [comment]
@blueorangutan package

**Comment by davidjumani** [comment]
NOTE: Destory VM is broken as it destorys the first vm in the list and not the one to be destroyed

File: src/views/AutogenView.vue [hunk 2]
```diff
@@ -686,6 +687,8 @@ export default {
       }
       this.currentAction = action
       this.currentAction.params = store.getters.apis[this.currentAction.api].params
+      this.resource = action.resource
+      this.$emit('change-resource', this.resource)
```
Comment (davidjumani) [inline src/views/AutogenView.vue:691]: @rhtyd Is there a better way to ensure the right resource gets passed to the action form? The other changes made in b02a796c10780c1d63430ae7c1ff381dcdf0e539 only apprear to work for autogenerated forms

**Comment by davidjumani** [comment]
@blueorangutan package

File: src/views/AutogenView.vue (hunk 2 above)
Comment (rhtyd) [reply src/views/AutogenView.vue:691]: I'm not sure we'll have to explore/test

**Comment by rhtyd** [comment]
Did a quick test LGTM, did not test for all possible cases; if there are issues we can restrict the quick view limited to only user facing views such as compute, storage, network, image etc.

File: src/components/view/ListView.vue [hunk 3]
```diff
@@ -61,18 +61,27 @@
...
 
     <span slot="name" slot-scope="text, record">
-      <div style="min-width: 120px">
-        <span v-if="$route.path.startsWith('/project')" style="margin-right: 5px">
-          <a-button type="dashed" size="small" shape="circle" icon="login" @click="changeProject(record)" />
-        </span>
-        <os-logo v-if="record.ostypename" :osName="record.ostypename" size="1x" style="margin-right: 5px" />
-        <console :resource="record" size="small" style="margin-right: 5px" />
+      <div style="min-width: 120px" >
+        <a-popover v-if="actions.length > 0" triggers="hover" placement="right">
```
Comment (rhtyd) [inline src/components/view/ListView.vue:65]: @davidjumani for read-only admin will this hide the name if there is nothing to be shown? the hover part LGTM but it's not clear perhaps add an explicit icon; what about components which don't have  name column as the first column?

**Comment by davidjumani** [comment]
@blueorangutan package

**Comment by davidjumani** [comment]
@blueorangutan package

**Comment by davidjumani** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@Pearl1594 can you review this thnx

**Comment by rhtyd** [comment]
@ravening can you help review and test this? Thanks.

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
Ping @utchoang @ravening can you test as well, thnx

**Comment by rhtyd** [comment]
@blueorangutan package

File: src/components/view/QuickView.vue [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  <a-popover v-if="enabled && actionsExist" triggers="hover" placement="right">
+    <template slot="content">
+      <action-button
+        :size="size"
+        :actions="actions"
+        :dataView="true"
+        :resource="resource"
+        @exec-action="execAction" />
+    </template>
+    <a-icon type="right-circle" style="margin-left: 5px" />
```
Comment (rhtyd) [inline src/components/view/QuickView.vue:28]: Could we use `<a-icon type="more" />` and  use float:right to keep it at right most side?

**Comment by rhtyd** [comment]
let's discuss/address the issue icon and if we should show icons with labels @davidjumani ?

File: src/components/view/QuickView.vue (hunk 4 above)
Comment (davidjumani) [reply src/components/view/QuickView.vue:28]: Using the more icon doesn't look as good, it shows up very faint
Keeping it float right means that the user has to go to the rightmost end of the column to get the hover and back to the record name to get into the detail view. This way the icon is right after the name

**Comment by davidjumani** [comment]
@blueorangutan package

**Pull request merged**